src/components/Pools/RemoveFarm/index.tsx

```diff
@@ -45,7 +45,18 @@ const RemoveFarm = ({ stakingInfo, version, onClose, onLoadingOrComplete, redire
 
   const mixpanel = useMixpanel();
 
-  const notAssociateTokens = useGetHederaTokenNotAssociated(rewardTokens || []);
+  const dummyPair = new Pair(
+    new TokenAmount(stakingInfo?.tokens?.[0], '0'),
+    new TokenAmount(stakingInfo?.tokens?.[1], '0'),
+    chainId,
+  );
+  const lpToken = dummyPair.liquidityToken;
+
+  const associatedTokens = rewardTokens || [];
+  // we need to check the lp token too
+  // because case a user farm in non Pangolin token/Wrapped token farm and compound to this farm
+  // in compoundTo
+  const notAssociateTokens = useGetHederaTokenNotAssociated([...associatedTokens, lpToken]);
```

Comment:
1. you are passing new array instance always which will re-run logic of `useGetHederaTokenNotAssociated`. instead you should do this
```ts
const tokensToCheck = useMemo(() => [...(rewardTokens || []), lpToken], [rewardTokens, lpToken]);
// we need to check the lp token too
// because case a user farm in non Pangolin token/Wrapped token farm and compound to this farm
// in compoundTo
const notAssociateTokens = useGetHederaTokenNotAssociated(tokensToCheck);
```
2. this is not working, it gives me invalid token id. as far as i can tell, `lpToken` is `pair contract address` but you need to pass `pgl fungible token` address. because of this logic it is always giving Associate PGL message even if PGL is associated.
3. this is how I am envisioning actual logic
```ts
const [pglToken] = useHederaPGLToken(stakingInfo?.tokens?.[0], stakingInfo?.tokens?.[1])
const tokensToCheck = useMemo(() => [...(rewardTokens || []), pglToken].filter(item => !!item), [rewardTokens, pglToken]) as Token[];
// we need to check the lp token too
// because case a user farm in non Pangolin token/Wrapped token farm and compound to this farm
// in compoundTo
const notAssociateTokens = useGetHederaTokenNotAssociated(tokensToCheck);
``` 
4. you need to update `useHederaPGLToken` so that it only runs for hedera chain and doesn't do api call for other chains kind of same as how we are doing it in `useGetAllHederaAssociatedTokens`.